lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -134,6 +147,62 @@ newTransactionLayer _proxy _protocolMagic epochLength = TransactionLayer
 
         pure $ toSealed $ SL.Tx unsigned addrWits scriptWits metadata
 
+    _mkDelegationJoinTx
+        :: PoolId
+        -> (k 'AddressK XPrv, Passphrase "encryption")
+        -> (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+        -> SlotId
+        -> [(TxIn, TxOut)]
+        -> [TxOut]
+        -> Either ErrMkTx (Tx, SealedTx)
+    _mkDelegationJoinTx poolId (accXPrv, pwd') keyFrom slot ownedIns outs = do
+        let timeToLive = (toSlotNo epochLength slot) + 7200
```

Comment:
I know this is still a draft, but I hope the similarities between `mkDelegationJoinTx` and `mkStdTx` can be factored out